log/namespaced/namespaced.go

```diff
@@ -0,0 +1,151 @@
+package namespaced
+
+import (
+	"sync"
+
+	"github.com/TheThingsNetwork/go-utils/log"
+)
+
+var NamespaceKey = "namespace"
+
+// Namespaced is a logger that only logs an entry when the namespace of that
+// entry is enabled
+type Namespaced struct {
+	sync.RWMutex
+	log.Interface
+	namespaces *ns
+	namespace  string
+}
+
+// WithNamespace adds a namespace to the logging context
+func WithNamespace(namespace string, ctx log.Interface) log.Interface {
+	return ctx.WithField(NamespaceKey, namespace)
+}
+
+// Wrap wraps the logger in a Namespaced logger and enables the specified
+// namespaces. See SetNamespaces for information on how to set the namspaces
+func Wrap(ctx log.Interface, namespaces ...string) *Namespaced {
+	return &Namespaced{
+		Interface: ctx,
+		namespaces: &ns{
+			namespaces: namespaces,
+		},
+	}
+}
+
+// SetNamespaces replaces the set of enabled namespaces
+// The namespaces follow this format:
+// "*"   enables all namespaces
+// "a"   enables namespace a
+// "-a"  disables namespace a (even if "a" or "*" is set)
+// For example:
```

Comment:
What about writing a function `UnsetNamespace` or `DisableNamespace` ?
In a shell that's pretty handy to use wildcards but in a golang program it might be better to be verbose and do a function for that.
What do you think ?